lib/api/objectDelete.js

```diff
@@ -73,6 +74,13 @@ function objectDelete(authInfo, request, log, cb) {
                     // versioning has been configured
                     return next(null, bucketMD, objMD);
                 }
+                // AWS only returns an object lock error if a version id
+                // is specified, else continue to create a delete marker
+                if (reqVersionId && isObjectLocked(
+                bucketMD, objMD, request.headers)) {
```

Comment:
As an alternative, could be divided like this for better readability:
```suggestion
                if (reqVersionId &&
                    isObjectLocked(bucketMD, objMD, request.headers)) {
```